resource/graph_node.go

```diff
@@ -570,6 +571,38 @@ func (w *GraphNode) resourceStatus() Status {
 	}
 }
 
+type graphNodeStats struct {
+	State    int
```

Comment:
Why not `NodeState`?

Reply:
Didn't think about this. But looking at the ultimate output, I think `State` is suitable. I agree the state is tracked by the "graph node", but it's really representing the resource. Here's a screenshot of how the "system name" + field name looks in FTDC graph form (not to be confused with "resource graph" and "graph node").

![image](https://github.com/user-attachments/assets/71839506-01e1-4b7f-bbd9-6dc9faf0a2f8)

Which reminds me I need to remember to escape underscores! Forgot to port that from the python I wrote:
```
            title = key.replace('_', '\\_')
```

Reply:
Let's discuss offline, but that graph is confusing to me.

Reply:
Discussed offline; didn't understand that line on the top right was a legend 😮‍💨 .

I'm still a little nervous about now having two "states" for a resource, one being `NodeState` and one being the reported state you have below based on the error returning from `w.Resource()`. Why would your states based on `err` be better than `NodeState`?

Reply:
Despite having a formal "node state" value, whether a resource is accessible (`func graphNode.Resource() (Resource, error)`) to me is the definitive "can a user call methods on this".

The [resource method](https://github.com/viamrobotics/rdk/blob/85f5b08edea7f65d27668359455dfed8f54dad73/resource/graph_node.go?plain=1#L143-L158), for better or worse, as implemented right now does not exhaust the node states. It only checks it for "removing".

I think if we're going to track one thing -- the output of "Resource" is the only thing that matters. If we want to supplement that with a second, more direct mapping of the `state NodeState` [value](https://github.com/viamrobotics/rdk/blob/85f5b08edea7f65d27668359455dfed8f54dad73/resource/graph_node.go?plain=1#L72), we can? I'd prefer to avoid tracking [each variable directly](https://github.com/viamrobotics/rdk/blob/85f5b08edea7f65d27668359455dfed8f54dad73/resource/graph_node.go?plain=1#L58-L67). But that's a straw-man argument, I know you're not actually asking for that.

Reply:
I buy the "can a user call methods on this" argument, but that just makes me think that the formal "node state" is not adequately representative of the state of the resource. I also agree that we shouldn't be tracking both the state you have here and `NodeState` (nor other fields on the graph node). 

I suppose I'm fine with what you have now, but I feel that `NodeState` probably needs to be representative of "can a user call methods on this" (cc @maximpertsov). Once that happens (is it not feasible for some reason I'm not understanding?), then I'd definitely prefer the FTDC-recorded state to be that `NodeState`.

Reply:
I'm on board with "as `.Resource()` changes, `Stats` shall too change".

Also I'm on board with just changing the names/values in the future if evidence suggests something else is better.